pandas/io/json/table_schema.py

```diff
@@ -273,10 +281,13 @@ def parse_table_schema(json, precise_float):
 
     Notes
     -----
-        Because ``write_json`` uses the string `index` to denote a name-less
-        ``Index``, this function sets the name of the returned ``DataFrame`` to
-        ``None`` when said string is encountered. Therefore, intentional usage
-        of `index` as the ``Index`` name is not supported.
+        Because :func:`DataFrame.to_json` uses the string 'index' to denote a
```

Comment:
how much would we break if we used something like what we do in parquet, e.g. ``__index_level_0__`` or something instead?

Reply:
cc @TomAugspurger

Reply:
yeah ok makes sense (and actually we could have back compat on this)

Reply:
https://github.com/pandas-dev/pandas/issues/19261